packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -252,25 +259,30 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     })
 
     const nonce = await this.signer.getTransactionCount()
```

Comment:
Getting the nonce both outside and inside of `customPopulateTransaction.appendSequencerBatch` seems like its prone to a race condition. Note that I do think its very unlikely in practice, but the function does take an optional second argument which may be able to take the nonce.

Reply:
Yes agreed, I just deleted the nonce logic